File: LearningProject/Models/Result.cs
namespace LearningProject.Models
{
    public class Result
    {
        public int Id { get; set; }
        public int UserExamId { get; set; }
        public int ExamVariantId { get; set; }
        public int Score { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        // Navigation properties
        public UserExam? UserExam { get; set; }
        public ExamVariant? ExamVariant { get; set; }
        public ICollection<AnswerResult>? AnswerResults { get; set; }
    }
}

File: client/src/app/core/models/Result.ts
import { AnswerResult } from "./AnswerResult";
import { ExamVariant } from "./ExamVariant";
import { UserExam } from "./UserExam";

export interface Result {
    id: number;
    userExamId: number;
    examVariantId: number;
    score: number;
    startDate: string;
    endDate: string;
    userExam?: UserExam | null;
    examVariant?: ExamVariant | null;
    answerResults?: AnswerResult[] | null;
  }
  